### PR TITLE
Feature/association viewmodel

### PR DESCRIPTION
**Pull request by ntissieres**
Creating the viewModel for the association

### CODE REVIEW

File: app/src/main/java/com/github/swent/echo/viewmodels/association/AssociationViewModel.kt [hunk 1]
```diff
@@ -0,0 +1,142 @@
...
+import com.github.swent.echo.data.model.Association
+import com.github.swent.echo.data.model.Event
+import com.github.swent.echo.data.repository.Repository
+import dagger.hilt.android.lifecycle.HiltViewModel
+import javax.inject.Inject
+import kotlinx.coroutines.flow.MutableStateFlow
+import kotlinx.coroutines.flow.asStateFlow
+import kotlinx.coroutines.launch
+
+enum class AssociationPage(var association: Association) {
```
Comment (Greenade) [inline app/src/main/java/com/github/swent/echo/viewmodels/association/AssociationViewModel.kt:16]: why do you need an empty association for every screen ?

Comment (ntissieres) [reply app/src/main/java/com/github/swent/echo/viewmodels/association/AssociationViewModel.kt:16]: I agree it's a bit weird but i didn't find a better solution. It's for DETAILS which need to take an Association as argument to know which association we want to display details

**Review by Greenade — CHANGES_REQUESTED** [review]
Nice work ! Not sure why you need en empty associations for every screen though ? Everything else looks good to me :)

File: app/src/main/java/com/github/swent/echo/viewmodels/association/AssociationViewModel.kt (hunk 1 above)
Comment (Greenade) [reply app/src/main/java/com/github/swent/echo/viewmodels/association/AssociationViewModel.kt:16]: Oh okay, I don't know if there is a better solution either 🤔 It's okay for now and we could always change it later

**Review by Greenade — APPROVED** [review]

**Comment by Greenade** [comment]
according to the discussion above, I removed the requested changes

**Pull request merged**